adrs/5541831634/README.md

```diff
@@ -0,0 +1,157 @@
+# React project files and directories names conversion
+
+* **State:** Draft
+* **Created:** 2020-12-01
+* **Tags:** react
+
+## Context
+
+Across the ecosystem, we can find all sorts of files and directory naming
+conventions in React projects.
+
+The variety of conventions causes misalignment and shifts focus away from more
+important matters.
+
+Although there are no silver bullets for this topic, committing to a convention
+will reduce cognitive load by triggering pattern recognition, enabling engineers
+to focus their brainpower on their tasks.
+
+## Resolution
+
+* You **must** follow [NodeJS files and directories names conversion](./../3122196229/README.md)
+* You **must** use `<rootDir>/src/@types` to store TypeScript definition files.
+* You **must** use `<rootDir>/src/app.tsx` to be your application root
+  component.
+
+### NextJS
+
+Some resolutions are related to NextJS projects.
+
+* You **must** use `<rootDir>/src/pages` for [NextJS Pages](https://nextjs.org/docs/basic-features/pages)
+  directory.
+* In NextJS project, you **must** re-export a NextJS app from
+  `<rootDir>/src/app.tsx` in `<rootDir>/src/pages/_app.tsx`
+* In NextJS project, you **must** re-export a NextJS Documents from
+  `<rootDir>/src/document.tsx` in `<rootDir>/src/pages/_document.tsx`
+* In NextJS project, you **must** re-export a NextJS Documents from
+  `<rootDir>/src/error.tsx` in `<rootDir>/src/pages/_error.tsx`
+
+### Fractal pattern directories and files
```

Comment:
Do you make a distinction between small, one-off micro frontends VS large SPA clients?   

Personally, I wonder if the need for the fractal pattern is a signal that you're shoving too much stuff into a given React app.  Another way to say that,  if you're building a small one-off client this directory structure seems like overkill.

Reply:
There are no differences between micro-frontend and large SPA,

Even when could be overkilling for a two-pages micro-frontend, the reduction of cognitive load still there, I rather have familiar code than go back to the problem statement.

This reminds me of Elixir projects vs Erlang folks, even when there is no point in adding `lib` directory and everything could be at the root (as Erlang does it), that one directory helps a shit ton to know where to start navigating the source code.

- Elders say: don't bother in adding a directory, no point, why that stupid lib directory.
- Millennials say: it matters to us, I do bother.

There is no right answer, all depends on your target audience.

That being said, objectively speaking, none of these matter to me, I can figure it out given enough time and resources, subjectively speaking. But that is exactly what I don't want, having to rely on people's experiences and having available resources to do my job.

Does that make sense?